src/ReactiveMP.jl

```diff
@@ -172,6 +172,26 @@ function __init__()
             return Flux.Optimise.update!(opt, vec(λ), vec(∇))
         end
     end
+
+    @require Zygote = "e88e6eb3-aa80-5325-afca-941959d7151f" begin
+        export ZygoteGrad
+
+        struct ZygoteGrad end
+
+        function compute_grad(::ZygoteGrad, A::F, vec_params) where {F}
+            Zygote.gradient(A, vec_params)[1]
+        end
+
+        function compute_hessian(::ZygoteGrad, A::G, ::F, vec_params) where {G, F}
+            Zygote.hessian(A, vec_params)
+        end
+
+        get_df_m(::ZygoteGrad, ::Type{<:UnivariateNormalNaturalParameters}, ::Type{<:UnivariateGaussianDistributionsFamily}, logp_nc::Function) =
```

Comment:
What is the purpose of `get_df_m`-like functions for Zygote? AFAIR these functions acted as a workaround specifically for the `ForwardDiff` library. If they are needed, there should be four functions, not two.

Also, as @bartvanerp mentioned already, the current implementation literally copy-pastes `compute_grad ` and `compute_hessian`. I think we discussed it already, but wouldn't it also be beneficial to compute gradient+hessian in one call (may be not a part of this PR though)?

Reply:
@bvdmitri I want to make our discussion clear for me, [the generic CVI step](https://github.com/biaslab/ReactiveMP.jl/blob/cvi-stability-fix/src/approximations/cvi.jl#L103-L119) is a 
```math
 \mathcal{F}^{-1}_{\lambda} \nabla_{\lambda} l
```
where $\mathcal{F}$ is the Fisher matrix, and $l$ is a likelihood. So we need the hessian of the [lognormalizer](https://github.com/biaslab/ReactiveMP.jl/blob/cvi-stability-fix/src/approximations/cvi.jl#L111) and gradient from the [likelihood](https://github.com/biaslab/ReactiveMP.jl/blob/cvi-stability-fix/src/approximations/cvi.jl#L109). I don't see how we can compute them at the same time.

So we can benefit from the computation of the hessian and the gradient only if the input message is a normal distribution. So you are proposing to create a function specifically for the normal case and call it kinda like `compute_df_dv`, am I correct?

Reply:
Yeah, in case if you take gradients of hessians of different things you cannot compute them together in one pass. I was assuming you compute it for one entity, but I was wrong.

Reply:
@bvdmitri @bartvanerp I added a bit more effective hessian/grad computation for [the normal case](https://github.com/biaslab/ReactiveMP.jl/blob/cvi-stability-fix/src/ReactiveMP.jl#L196-L200).

Its working time is same. But it allocates 10% less.

You can reproduce my benchmark with this code:
```julia

using ReactiveMP
using BenchmarkTools
using ForwardDiff

struct EmptyOptimizer end

struct ForwardDiffOld end

function ReactiveMP.compute_gradient(::ForwardDiffOld, f::F, vec_params) where {F}
    ForwardDiff.gradient(f, vec_params)
end

function ReactiveMP.compute_hessian(::ForwardDiffOld, f::F, vec_params) where {F}
    ForwardDiff.hessian(f, vec_params)
end

meta = CVI(1, 100, EmptyOptimizer())
meta_old = CVI(1, 100, EmptyOptimizer(), ForwardDiffOld())

function sum_square(vec)
    return sum(map(x -> x^2, vec))
end

@benchmark ReactiveMP.compute_df_mv(meta_old, sum_square, $rand(20))
@benchmark ReactiveMP.compute_df_mv(meta, sum_square, $rand(20))
```